integration/test/performancetest/performance_test.go

```diff
@@ -1,59 +1,158 @@
+	
+				test.StartAgent(configOutputPath, true)
+	
+				agentRunDuration := agentRuntimeMinutes * time.Minute
+	
+				StartLogWrite(t, agentRunDuration, configPath, tps)
+	
+				log.Printf("Agent has been running for : %s\n", (agentRunDuration).String())
+				test.StopAgent()
+	
+				//collect data
+				data, err := GetPerformanceMetrics(instanceId, agentRuntimeMinutes, agentContext, configPath)
+				//@TODO check if metrics are zero remove them and make sure there are non-zero metrics existing
+				if err != nil {
+					log.Println("Error: ", err)
+					t.Fatalf("Error: %v", err)
+				}
+	
+				if data == nil {
+					t.Fatalf("No data")
+				}
+				
+				_, err = dynamoDB.SendItem(data)
+				if err != nil {
+					t.Fatalf("Error: couldnt upload metric data to table")
+				}
+			})
+		}
+	} 
+}
+
+//StartLogWrite starts go routines to write logs to each of the logs that are monitored by CW Agent according to
+//the config provided
+func StartLogWrite(t *testing.T, agentRunDuration time.Duration, configFilePath string, tps int) {
+	//create wait group so main test thread waits for log writing to finish before stopping agent and collecting data
+	var logWaitGroup sync.WaitGroup
 
-	test.StartAgent(configOutputPath, true)
+	logPaths := GetLogFilePaths(configFilePath)
 
-	agentRunDuration := agentRuntimeMinutes * time.Minute
-	//let agent run before collecting performance metrics on it
-	time.Sleep(agentRunDuration)
-	log.Printf("Agent has been running for : %s\n", (agentRunDuration).String())
-	test.StopAgent()
+	for i := 0; i < len(logPaths); i++ {
+		logWaitGroup.Add(1)
+		go func() {
+			defer logWaitGroup.Done()
+			WriteToLogs(t, logPaths[i], agentRunDuration, tps)
+		}
+	}
+
+	//wait until writing to logs finishes
+	logWaitGroup.Wait()
+}
 
-	//collect data
-	data, err := GetPerformanceMetrics(instanceId, agentRuntimeMinutes, agentContext)
-	//@TODO check if metrics are zero remove them and make sure there are non-zero metrics existing
+//WriteToLogs opens a file at the specified file path and writes the specified number of lines per second (tps)
+//for the specified duration
+func WriteToLogs(t *testing.T, filePath string, durationMinutes time.Duration, tps int) {
+	f, err := os.Create(filePath)
 	if err != nil {
-		log.Println("Error: ", err)
-		t.Fatalf("Error: %v", err)
+		t.Fatalf("Error occurred creating log file for writing: %v", err)
 	}
+	defer f.Close()
+	defer os.Remove(filePath)
+
+	log.Printf("Writing lines to %s with %d transactions per second", filePath, tps)
+
+	startTime := time.Now()
 
-	//------Placeholder to put data into database------//
-	//useless code so data get used and compiler isn't mad
-	if data == nil {
-		t.Fatalf("No data")
+	ticker := time.Ticker(1 * time.Second)
+	defer ticker.Stop()
+
+	//loop until the test duration is reached
+	for {
+		select {
+		case <-ticker.C:
+			for i := 0; i < tps; i++ {
+				_, err = f.WriteString(fmt.Sprintf("%s - #%d This is a log line.\n", currTime.Format(time.StampMilli), i))
```

Comment:
If `f.WriteString` fails the first time, should it fail the next try too? I can't imaging the scenario which one fails and others pass. Therefore, would it be better to `t.Fatalf`?

Reply:
I will change this